src/sentry/processing/screenshot_processor.cpp

```diff
@@ -0,0 +1,79 @@
+#include "screenshot_processor.h"
+
+#include "sentry/common_defs.h"
+#include "sentry/util/print.h"
+#include "sentry/util/screenshot.h" // TODO: incorporate
+#include "sentry_options.h"
+
+#include <godot_cpp/classes/dir_access.hpp>
+#include <godot_cpp/classes/display_server.hpp>
+#include <godot_cpp/classes/engine.hpp>
+#include <godot_cpp/classes/file_access.hpp>
+#include <godot_cpp/classes/os.hpp>
+
+Ref<SentryEvent> ScreenshotProcessor::process_event(const Ref<SentryEvent> &p_event) {
+	if (p_event.is_null()) {
+		sentry::util::print_error("internal error: can't process null event");
+		return nullptr;
+	}
+
+	bool is_main_thread = OS::get_singleton()->get_thread_caller_id() == OS::get_singleton()->get_main_thread_id();
+
+	{
+		std::lock_guard lock{ mutex };
+
+		int32_t current_frame = Engine::get_singleton()->get_frames_drawn();
+		if (current_frame == last_screenshot_frame) {
+			sentry::util::print_debug("skipping screenshot");
+			// Screenshot already exists for this frame — nothing to do.
+			return p_event;
+		}
+		if (is_main_thread) {
+			last_screenshot_frame = current_frame;
```

Comment:
I'm thinking this is done too early - e.g. if we exit early in one of the `ifs` below, like get_level or before_capture checks, we'd still memoize the `last_screenshot_frame` here although the screenshot hasn't been taken.